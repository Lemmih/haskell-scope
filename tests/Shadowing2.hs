module Shadowing2 where

y = y

x y = y
  where
    y = 10
