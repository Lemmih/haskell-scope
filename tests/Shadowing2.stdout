Scope errors:

Definitions:
  Definition 1, from Shadowing2.decl0:
    [4my[0m = y
  Definition 2, from Shadowing2.decl1:
    [4mx y = y[0m
    [4m  where[0m
    [4m    y = 10[0m
  Definition 3, from Shadowing2.decl1.match0:
    x [4my[0m = y
  Definition 4, from Shadowing2.decl1.match0.rhs.binds0:
        [4my[0m = 10

Use sites:
  Definition used: 1
    y = [4my[0m
  Definition used: 2
    [4mx[0m y = y
  Definition used: 4
    x y = [4my[0m

Exports:
  x, Value from Shadowing2
  y, Value from Shadowing2
